improver_tests/calibration/reliability_calibration/test_ApplyReliabilityCalibration.py

```diff
@@ -230,6 +230,143 @@ def test_exception_without_relative_to_threshold(self):
             self.plugin._ensure_monotonicity(self.forecast)
 
 
+class Test__combine_undersampled_bins(unittest.TestCase):
+
+    """Test the _combine_undersampled_bins method."""
+
+    def test_no_undersampled_bins(self):
+        """Test no bins are combined when no bins are under-sampled."""
+        obs_count = np.array([0, 250, 500, 750, 1000], dtype=np.float32)
+        forecast_probability_sum = np.array([0, 250, 500, 750, 1000], dtype=np.float32)
+        forecast_count = np.array([1000, 1000, 1000, 1000, 1000], dtype=np.float32)
+
+        result = Plugin()._combine_undersampled_bins(
+            obs_count, forecast_probability_sum, forecast_count
+        )
+
+        assert_array_equal(
+            result, [obs_count, forecast_probability_sum, forecast_count]
+        )
+
+    def test_one_undersampled_bin_at_top(self):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/metoppv_tech/improver/pullRequest?prid=6014128)